### PR TITLE
Bump mapboxEvents, mapboxCore and Robolectric dependency versions to 6.0.0, 3.0.0 and 4.3.1, targetSdkVersion and compileSdkVersion to 29 and remove buildToolsVersion references

### PR DESCRIPTION
**Pull request by Guardiola31337**
## Description

Bumps `mapbox-android-telemetry`, `mapbox-android-core` and `robolectric` versions to `6.0.0`, `3.0.0` and `4.3.1` respectively, `targetSdkVersion` and `compileSdkVersion` to `29`, remove unnecessary `buildToolsVersion` references as the project automatically uses a default version of the build tools that the plugin specifies, `29` in this case after bumping, workaround API 29 / Java 9 Robolectric issues forcing API 28 and remove Robolectric from tests that do not need it

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

Getting the latest versions from `mapboxEvents`, `mapboxCore` and `robolectric` including the new features and bug fixes and update `targetSdkVersion` and `compileSdkVersion` to the latest stable `29`

### Implementation

- Bump `mapboxEvents` and `mapboxCore` to `6.0.0`, `3.0.0` respectively in `dependencies.gradle`
- Bump `robolectric` to `4.3.1` in `dependencies.gradle`
- Bump `targetSdkVersion` and `compileSdkVersion` in `dependencies.gradle`
- Remove `buildToolsVersion` from `dependencies.gradle` and all the references to it
- https://github.com/mapbox/mapbox-navigation-android/pull/3230#issuecomment-649081774
- Remove Robolectric from tests that do not need it https://github.com/mapbox/mapbox-navigation-android/issues/3221#issuecomment-649112480 https://github.com/mapbox/mapbox-navigation-android/pull/3230#issuecomment-649095920

## Testing

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR
- [ ] We might need to update / push `api/current.txt` files after running `$> make 1.0-core-update-api` (Core) / `$> make 1.0-ui-update-api` (UI) if there are changes / errors we're 🆗 with (e.g. `AddedMethod` changes are marked as errors but don't break SemVer) 🚀 If there are SemVer breaking changes add the `SEMVER` label. See [Metalava](https://github.com/mapbox/mapbox-navigation-android/blob/master/docs/metalava.md) docs

cc @nkukday @harvsu

### CODE REVIEW

**Comment by Guardiola31337** [comment]
CI ❌ are caused because of bumping `targetSdkVersion` and `compileSdkVersion` to `29` and  Robolectric `4.3.1` version that supports API level `29` https://github.com/robolectric/robolectric/issues/5207#issuecomment-541795921 requiring Java 9 https://github.com/robolectric/robolectric/issues/5258#issuecomment-556282605 and the Docker image `android-ndk-r21:latest` doesn't include it 👀 https://github.com/mapbox/mapbox-navigation-android/pull/2557#issuecomment-649073921

Going to workaround it until we have a new Docker image with Java 9 adding a `robolectric.properties` file inside `test/resources` modules directories with the following line:

```
sdk=28
```

so that Robolectric is forced to use API 28 instead of 29.

**Comment by Guardiola31337** [comment]
Currently blocked by https://github.com/mapbox/mapbox-navigation-android/issues/3221#issuecomment-649093074 

cc @korshaknn

**Comment by Guardiola31337** [comment]
> Currently blocked by https://github.com/mapbox/mapbox-navigation-android/issues/3221#issuecomment-649093074 

It turns out that Robolectric is not needed at all in `MapboxHybridRouterTest` https://github.com/mapbox/mapbox-navigation-android/issues/3221#issuecomment-649112480

Pushed 👇 

> - Remove Robolectric from tests that do not need it https://github.com/mapbox/mapbox-navigation-android/issues/3221#issuecomment-649112480 https://github.com/mapbox/mapbox-navigation-android/pull/3230#issuecomment-649095920

We should be good now. Ready for 🚀 

cc @korshaknn @kmadsen

File: libnavigator/src/test/resources/robolectric.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+sdk=28
```
Comment (kmadsen) [inline libnavigator/src/test/resources/robolectric.properties:1]: @Guardiola31337 can tell this is on purpose. but is there an option to put this in one place?

libtesting-utils or libtesting-ui? or could we have the number in one place?

Comment (kmadsen) [reply libnavigator/src/test/resources/robolectric.properties:1]: can robolectric.properties see dependencies.gradle?

File: libdirections-hybrid/src/test/java/com/mapbox/navigation/route/hybrid/MapboxHybridRouterTest.kt [hunk 2]
```diff
@@ -22,12 +22,9 @@ import kotlinx.coroutines.delay
 import kotlinx.coroutines.runBlocking
 import org.junit.Before
 import org.junit.Test
-import org.junit.runner.RunWith
-import org.robolectric.RobolectricTestRunner
 
 @InternalCoroutinesApi
 @ExperimentalCoroutinesApi
-@RunWith(RobolectricTestRunner::class)
```
Comment (kmadsen) [inline libdirections-hybrid/src/test/java/com/mapbox/navigation/route/hybrid/MapboxHybridRouterTest.kt:30]: oh we can just delete 💯

File: app/src/main/java/com/mapbox/services/android/navigation/testapp/example/ui/ExampleViewModel.kt [hunk 3]
```diff
@@ -181,7 +181,7 @@ class ExampleViewModel(application: Application) : AndroidViewModel(application)
                         R.string
                             .default_route_profile
                     )
-                )
+                ) ?: context.getString(R.string.default_route_profile)
```
Comment (kmadsen) [inline app/src/main/java/com/mapbox/services/android/navigation/testapp/example/ui/ExampleViewModel.kt:184]: 🔨

File: libnavigator/src/test/resources/robolectric.properties (hunk 1 above)
Comment (Guardiola31337) [reply libnavigator/src/test/resources/robolectric.properties:1]: Those are great questions, do you know? 😅

**Comment by Guardiola31337** [comment]
Going ahead and merging here 🚀

**Pull request merged**